File: 2015/01-2.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
	FILE *filePointer;
	char ch;
 	int floor = 0;

	const int up =  *"(";
	const int down = *")";

	filePointer = fopen("01.txt", "r");

    int pos = 0;
	while ((ch = fgetc(filePointer)) != EOF)
	{
        pos++;
		if (ch == up) {
			floor++;
		} else if (ch == down) {
			floor--;
		}

        if (floor == -1) {
            printf("%d\n", pos);
            return 0;
        }

	}
	return 0;
}

File: 2015/01-2.py
floor = 0
pos = 0
with open('01.txt') as fp:
    while True:
        char = fp.read(1)
        if not char:
            break
        pos += 1
        if char == "(":
            floor += 1
        elif char == ")":
            floor -= 1

        if floor == -1:
            break
print(pos)
